lib/vanagon/component.rb

```diff
@@ -231,16 +233,25 @@ def get_build_dir
     # Fetches secondary sources for the component. These are just dumped into the workdir currently.
     #
     # @param workdir [String] working directory to put the source into
-    def get_sources(workdir)
+    def get_sources(workdir) # rubocop:disable Metrics/AbcSize
       sources.each do |source|
         src = Vanagon::Component::Source.source(
           source.url, workdir: workdir, ref: source.ref, sum: source.sum
         )
         src.fetch
         src.verify
+        # set src.file to only be populated with the basename instead of entire file path
+        src.file = File.basename(src.url)
+        extract_with << src.extract(platform.tar) if src.respond_to? :extract
       end
```

Comment:
so get_source is used for the primary source which ends up getting some special treatment as opposed to the secondary sources which are stored in get_sources.  The fact that they're being treated differently now doesn't mean they should be, but unraveling the differences that matter and ones that can be changed feels like it could be a huge yak.

For example, the primary source (get_source) is set via the project url, and all of the secondary are set via add_source.

I don't think we should block this work on figuring this out, but we can and should absolutely add a ticket in the VANAGON project.